code/modules/admin/admin.dm

```diff
@@ -425,19 +425,20 @@
 	var/list/options = list("Regular Restart", "Hard Restart (No Delay/Feeback Reason)", "Hardest Restart (No actions, just reboot)")
 	if(world.RunningService())
 		options += "Service Restart (Force restart DD)";
-	var result = input(usr, "Select reboot method", "World Reboot", options[1]) as null|anything in options
-	if(result)
-		SSblackbox.add_details("admin_verb","Reboot World") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
-		switch(result)
-			if("Regular Restart")
-				SSticker.Reboot("Initiated by [usr.client.holder.fakekey ? "Admin" : usr.key].", "admin reboot - by [usr.key] [usr.client.holder.fakekey ? "(stealth)" : ""]", 10)
-			if("Hard Restart (No Delay, No Feeback Reason)")
-				world.Reboot()
-			if("Hardest Restart (No actions, just reboot)")
-				world.Reboot(fast_track = TRUE)
-			if("Service Restart (Force restart DD)")
-				GLOB.reboot_mode = REBOOT_MODE_HARD
-				world.ServiceReboot()
+	if(alert(usr, "This will end the round, are you SURE you want to do this? [SSticker.admin_delay_notice ? "An admin has already delayed the round end for the following reason: [SSticker.admin_delay_notice]" : ""]", "Confirmation", "Yes", "No") == "Yes")

```

Comment:
This message shouldn't pop up if there's no delay notice, as the input already gives you a cancel option